tests/integration_tests/test_public_api.py

```diff
@@ -0,0 +1,144 @@
+import ipaddress
+from contextlib import contextmanager, suppress
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Generator, Type
+
+import pytest
+
+import pycloudlib
+from pycloudlib.cloud import BaseCloud
+from pycloudlib.instance import BaseInstance
+from pycloudlib.util import LTS_RELEASES
+
+cloud_config = """\
+#cloud-config
+runcmd:
+  - echo 'hello' > /var/tmp/example.txt
+"""
+
+
+@pytest.fixture
+def cloud(request):
+    cloud_instance: BaseCloud = request.param(
+        tag="pycloudlib-integration",
```

Comment:
Let's trim down the tag for readability. I also think there are character limits on some clouds for instance name which gets derived from cloud.tag in some cases.